docido_sdk/toolbox/date_ext.py

```diff
@@ -26,13 +46,94 @@ def feeling_lucky(cls, obj):
         elif isinstance(obj, datetime):
             return cls.from_datetime(obj)
         else:
-            raise ValueError("Don't know how to get timestamp")
+            raise ValueError(
+                u"Don't know how to get timestamp from '{}'".format(obj)
+            )
+
+    @classmethod
+    def check_mispelled_day(cls, timestr):
+        day_extraction = cls.START_WITH_DAY_OF_WEEK.match(timestr)
+        if day_extraction is not None:
+            day = day_extraction.group(1).lower()
+            if len(day) == 3:
+                dataset = DAYS_ABBR
+            else:
+                dataset = DAYS
+            if day not in dataset:
+                days = list(dataset)
+                days.sort(key=lambda e: levenshtein(day, e))
+                return days[0] + day_extraction.group(2)
+
+    @classmethod
+    def remove_parenthesis_around_tz(cls, timestr):
+        """get rid of parenthesis around timezone: (GMT) => GMT"""
+        parenthesis = cls.TIMEZONE_PARENTHESIS.match(timestr)
+        if parenthesis is not None:
+            return parenthesis.group(1) + parenthesis.group(2)
+
+    @classmethod
+    def remove_quotes_around_tz(cls, timestr):
+        quoted = cls.QUOTED_TIMEZONE.match(timestr)
+        if quoted is not None:
+            return quoted.group(1) + quoted.group(2)
+
+    @classmethod
+    def remove_timezone(cls, timestr):
+        if re.match(r".*[\-+]?\d{2}:\d{2}$", timestr):
+            return re.sub(
+                r"(.*)(\s[\+-]?\d\d:\d\d)$",
+                r"\1",
+                timestr
+            )
+
+    @classmethod
+    def fix_timezone_separator(cls, timestr):
+        tz_sep = cls.TIMEZONE_SEPARATOR.match(timestr)
+        if tz_sep is not None:
+            return tz_sep.group(1) + tz_sep.group(2) + ':' + tz_sep.group(3)
+        return timestr
 
     @classmethod
-    def from_str(cls, timestr):
+    def from_str(cls, timestr, shaked=False):
```

Comment:
@tuxnco I prefer `shaked` because `checked` would assume that the patched date is valid. But here, in the first `except` clause, the modified date can still be crap even after applying the *checkers*.

Actually this is the `checker` variable that I should rename to `shaker`.